notebooks/224-stable-diffusion-text-to-image/224-stable-diffusion-text-to-image.ipynb

```diff
@@ -0,0 +1,1155 @@
+        "# to make sure that model works and save input for vae decoder\n",
+        "output_latents = pipe.unet(latents, t, encoder_hidden_state)[0]\n",
+        "\n",
+        "with torch.no_grad():\n",
+        "    torch.onnx.export(\n",
+        "        pipe.unet, \n",
+        "        (latents, t, encoder_hidden_state), 'unet/unet.onnx',\n",
+        "        input_names=['sample', 'timestep', 'encoder_hidden_state'],\n",
+        "        output_names=['out_sample']\n",
+        "    )\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### VAE\n",
+        "\n",
+        "The VAE model has two parts, an encoder and a decoder. The encoder is used to convert the image into a low dimensional latent representation, which will serve as the input to the U-Net model. The decoder, conversely, transforms the latent representation back into an image.\n",
+        "\n",
+        "During latent diffusion training, the encoder is used to get the latent representations (latents) of the images for the forward diffusion process, which applies more and more noise at each step. During inference, the denoised latents generated by the reverse diffusion process are converted back into images using the VAE decoder. As we will see during inference we **only need the VAE decoder**.\n",
+        "\n",
+        "In our inference pipeline, we will need only decoding part of VAE, but forward function, which used for tracing runs encoding too. For obtaining only necessary part of model, we will wrap it into new model class."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "fQvd38qPHLtq"
+      },
+      "outputs": [],
+      "source": [
+        "class VAEDecoderWrapper(torch.nn.Module):\n",
+        "    def __init__(self, vae):\n",
+        "        super().__init__()\n",
+        "        self.vae = vae\n",
+        "\n",
+        "    def forward(self, latents):\n",
+        "        return self.vae.decode(latents)\n",
+        "\n",
+        "\n",
+        "vae_decoder = VAEDecoderWrapper(pipe.vae)\n",
+        "\n",
+        "latents_uncond, latents_text = output_latents[0].unsqueeze(0), output_latents[1].unsqueeze(0)\n",
+        "latents = latents_uncond + 7.5 * (latents_text - latents_uncond)\n",
+        "\n",
+        "vae_decoder.eval()\n",
+        "with torch.no_grad():\n",
+        "    torch.onnx.export(vae_decoder, latents, 'vae_decoder.onnx', input_names=['latents'], output_names=['sample'])"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# to reduce memory consumption, remove pipeline form memory as we will not use it anymore\n",
+        "del pipe"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Convert models using OpenVINO Model Optimizer API"
```

Comment:
I would call it "Convert models to OpenVINO IR" regardless of what API is used underneath. It can be clarified in the text that explains this step. 
BTW., why do we need two separate step here, like "Convert to ONNX" and "Convert to IR"? ONNX is just an intermediate stage in the route from PyTorch to OV IR. It can be eliminated when we have PyTorch frontend. @slyalin, what do you think? My suggestion is to combine these step into one and call it "Convert models to OpenVINO IR".

Reply:
I wanted to highlight that some pre-work before conversion like onnx conversion, model freezing for tf, convert dynamic models to static for paddle can be required, it is reason why onnx conversion is separated as its own step.
Of course, with torch frontend readiness or possibly with mo.convert phase 2(onnx conversion will be run internally) it will go. However, regarding mo.convert case, it is unclear for me now regarding custom symbolic function registration for `broadcast_to` (possibly I need to make some hucks with replacement to expand directly in torch or use onnx fe extensions on our side (just convert to onnx with preserving custom op)

Reply:
@AlexKoff88 so, should I combain onnx export and mo in one section?

Reply:
sections merged